hotel-reservation-system/src/components/User/UserTable.js

```diff
@@ -40,36 +50,67 @@ export default function UserTable({ hotelId }) {
   const [userId, setUserId] = useState();
   const [addUserDialogOpen, setAddUserDialogOpen] = useState(false);
   const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+  const [alertOpen, setAlertOpen] = useState(false);
+  const [alertMessage, setAlertMessage] = useState("");
+  const [userEmail, setUserEmail] = useState("");
+  const [userSurname, setUserSurname] = useState("");
+  const [currentSortField, setCurrentSortField] = useState("");
+  const [currentAscending, setCurrentAscending] = useState("");
+  const [alertSuccessStatus, setAlertSuccessStatus] = useState(true);
   const form = <Register handleClose={handleCloseAddUserDialog}></Register>;
 
   useEffect(() => {
-    const loadUsers = async () => {
-      await API.get(
-        "/users/" +
-          "?PageNumber=" +
-          pageForRequest +
-          "&PageSize=" +
-          rowsPerPage,
-        {
-          headers: { Authorization: "Bearer " + token },
-        }
-      )
-        .then((response) => response.data)
-        .then((data) => {
-          setUsers(data.items);
-          setMaxNumberOfUsers(data.numberOfItems);
-        })
-        .catch((error) => console.log(error.response.data.message));
-    };
-    if (
-      deleteDialogOpen === false &&
-      addUserDialogOpen === false &&
-      userId === undefined
-    ) {
+    if (deleteDialogOpen === false && addUserDialogOpen === false) {
       loadUsers();
     }
   }, [rowsPerPage, page, deleteDialogOpen, addUserDialogOpen]);
 
+  const loadUsers = async (email, surname, flag, sortField, ascending) => {
+    if (flag === undefined) {
+      email = userEmail;
+      surname = userSurname;
+    }
+    if (email === undefined) {
+      email = "";
+    }
+    if (surname === undefined) {
+      surname = "";
+    }
+    if (sortField === null || sortField === undefined) {
+      sortField = currentSortField;
+    }
+    if (ascending === null || ascending === undefined) {
+      ascending = currentAscending;
+    }
+    if (ascending === "asc") {
+      ascending = true;
+    } else {
+      ascending = false;
+    }
+    await API.get(
+      "/users",
+      {
+        params: {
+          Email: undefined,
+          Surname: surname,
+          PageNumber: pageForRequest,
+          PageSize: rowsPerPage,
+          SortField: sortField,
+          Ascending: ascending,
```

Comment:
Maybe Ascending: (ascending ||  currentAscending) === "asc"

Reply:
changed to this construction in all places